src/sentry/incidents/endpoints/serializers/workflow_engine_detector.py

```diff
@@ -1,34 +1,325 @@
+                )
+                result[detector]["latestIncident"] = serialized_group_open_period
+
+    def get_attrs(
+        self, item_list: Sequence[Detector], user: User | RpcUser | AnonymousUser, **kwargs: Any
+    ) -> defaultdict[Detector, dict[str, Any]]:
+        detectors = {item.id: item for item in item_list}
+        result: DefaultDict[Detector, dict[str, Any]] = defaultdict(dict)
+
+        detector_workflow_condition_group_ids = [
+            detector.workflow_condition_group.id
+            for detector in detectors.values()
+            if detector.workflow_condition_group
+        ]
+        detector_trigger_data_conditions = DataCondition.objects.filter(
+            condition_group__in=detector_workflow_condition_group_ids,
+            condition_result__in=[DetectorPriorityLevel.HIGH, DetectorPriorityLevel.MEDIUM],
+        )
+
+        action_filter_data_condition_groups = DataCondition.objects.filter(
+            comparison__in=[
+                detector_trigger.condition_result
+                for detector_trigger in detector_trigger_data_conditions
             ],
-            "dateModified": datetime.now(),
-            "dateCreated": datetime.now(),
-            "createdBy": {},
-            "description": "test",
-            "detectionType": "test",
+        ).exclude(condition_group__in=detector_workflow_condition_group_ids)
+
+        dcgas = DataConditionGroupAction.objects.filter(
+            condition_group__in=[
+                action_filter.condition_group
+                for action_filter in action_filter_data_condition_groups
+            ]
+        ).select_related("action")
+        actions = [dcga.action for dcga in dcgas]
+
+        # add sentry app data
+        organization_id = [detector.project.organization_id for detector in detectors.values()][0]
+        sentry_app_installations_by_sentry_app_id = (
+            self.add_sentry_app_installations_by_sentry_app_id(actions, organization_id)
+        )
+
+        # add trigger and action data
+        serialized_data_conditions: list[dict[str, Any]] = []
+        for trigger in detector_trigger_data_conditions:
+            serialized_data_conditions.extend(
+                serialize(
+                    [trigger],
+                    user,
+                    WorkflowEngineDataConditionSerializer(),
+                    **kwargs,
+                )
+            )
+        self.add_triggers_and_actions(
+            result,
+            detectors,
+            sentry_app_installations_by_sentry_app_id,
+            serialized_data_conditions,
+        )
+        self.add_projects(result, detectors)
+        self.add_created_by(result, list(detectors.values()))
+        self.add_owner(result, list(detectors.values()))
+        # skipping snapshot data
+
+        if "latestIncident" in self.expand:
+            # the most horrible way to map a detector to it's action ids but idk if I can make this less horrible
```

Comment:
I realize this is ... not great, but there doesn't seem to be an easier way to keep track of this information.